crawler/crawl-device.py

```diff
@@ -387,16 +397,30 @@ def database_push(username, up_hosts, device_types, os_names, os_versions, city,
             
         except Exception as e:
             print("Error:", e)
-            
+
+'''
+Function: check_args()
+Args: None
+Checks arguments provided for a username
+Returns username field
+'''
+def check_args():
+    if (len(sys.argv) != 2):
+        print("Usage: python3 crawl-device.py <username>")
+        raise ValueError("Exactly one argument is required.")
+    username = sys.argv[1]
+    print(f"Username: {username}")
+    return username
+
+
 '''
 Main Program Driver
 '''
 if __name__ == "__main__":
     print("Started Crawler...")
-    
     # Username is a cmd-line argument passed from UI to script
-    # TODO: Require a cmd-line argument to create DB API with scope given username
-    username = ""
+    # Require a cmd-line argument to create DB API with proper scope
+    username = check_args()
```

Comment:
I would consider changing the name of the `check_args()` method, or doing something to make the "english reading" of the code clearer; currently, it feels clunky to me to read "username gets set to the return value of check_args"... But this is minor, and doesn't really need to be changed unless you want to bother